components/api/db/queries.bal

```diff
@@ -230,3 +230,14 @@ public const string GET_ARTIFACT_IMAGE_ID = "SELECT ARTIFACT_IMAGE_ID FROM REGIS
 
 public const string GET_ARTIFACT_COUNT = "SELECT COUNT(ARTIFACT_ID) FROM REGISTRY_ARTIFACT WHERE ARTIFACT_IMAGE_ID = ?
                                     AND VERSION LIKE ?";
+
+public const string DELETE_ARTIFACT_QUERY = "DELETE FROM REGISTRY_ARTIFACT WHERE ARTIFACT_IMAGE_ID = (SELECT ARTIFACT_IMAGE_ID FROM
```

Comment:
Anybody who has push access should be allowed to delete an artifact or delete an image . @nadundesilva  WDYT ?. 

Basically push rights means write permission to image. Deleting is a form of writing.